core/shared/src/main/scala/spire/math/Interval.scala

```diff
@@ -767,6 +767,48 @@ sealed abstract class Interval[A](implicit order: Order[A]) extends Serializable
 
   def foldOver[B](init: B, step: A)(f: (B, A) => B)(implicit ev: AdditiveMonoid[A], nt: NumberTag[A]): B =
     iterator(step).foldLeft(init)(f)
+
+  /**
+    * Result of overlapping this interval with another one.
+    * Can be one of the following:
+    * - [[spire.math.interval.Equals]] if intervals are equal
+    * - [[spire.math.interval.StrictlyLess]] if intervals are notEmpty don't intersect
```

Comment:
The alternative is that we add 2 more cases to deal with `GreaterAndOverlap` and `StrictlyGreater` and we respect the original order of the args. In this case, we can directly use the result to determine the ordering of the intervals.